### ISSUE TITLE
Should Markdown Transform depend on Slate at all?

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/markdown-transform#116]
We aren't using Slate in this package, except on one line:
https://github.com/accordproject/markdown-transform/blob/aa954423b4557e4b2714e13db728761866a60d09/packages/markdown-slate/src/SlateTransformer.js#L54

Unless we decide to validate against a Slate schema, I see little benefit of that additional dependency. We could make that call rely on the Slate JSON representation and let users load to Slate from it instead.

This would also let us remove of the `slate` and `immutable` peer dependencies.

**Comment by jeromesimeon** [comment]
@DianaLease @dselman

**Comment by DianaLease** [comment]
Interesting. It looks like we are using that `fromCiceroMark` method in `markdown-cli`. Would this change affect anything there? And now that I see that, I'm wondering if that means Slate needs to be a peer dependency of `markdown-cli` since `markdown-cli` depends on `markdown-slate`, or if those peer dependencies get bubbled up.

**Comment by jeromesimeon** [comment]
> Interesting. It looks like we are using that `fromCiceroMark` method in `markdown-cli`. Would this change affect anything there? And now that I see that, I'm wondering if that means Slate needs to be a peer dependency of `markdown-cli` since `markdown-cli` depends on `markdown-slate`, or if those peer dependencies get bubbled up.

Ouch. Good catch @DianaLease ! This indeed breaks `markdown-cli`

```
bash-3.2$ npm install -g @accordproject/markdown-cli 
/Users/jeromesimeon/.nvm/versions/node/v8.12.0/bin/markus -> /Users/jeromesimeon/.nvm/versions/node/v8.12.0/lib/node_modules/@accordproject/markdown-cli/index.js
npm WARN @accordproject/markdown-slate@0.7.6 requires a peer of slate@0.47.x but none is installed. You must install peer dependencies yourself.
npm WARN @accordproject/markdown-slate@0.7.6 requires a peer of immutable@>=3.8.1 || >4.0.0-rc but none is installed. You must install peer dependencies yourself.

+ @accordproject/markdown-cli@0.7.6
removed 442 packages and updated 5 packages in 20.128s
bash-3.2$ markus --version
module.js:550
    throw err;
    ^

Error: Cannot find module 'slate'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/jeromesimeon/.nvm/versions/node/v8.12.0/lib/node_modules/@accordproject/markdown-cli/node_modules/@accordproject/markdown-slate/lib/SlateTransformer.js:529:38)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
```

I would like to make the proposed change and remove the peer dependencies.

**Comment by jeromesimeon** [comment]
Update on this. All the testing assumes that the Slate document has been normalized / round tripped through real Slate DOM.

That makes me feel less clear that removing the Slate dependency is the right thing to do. I'm leaving this issue open for review and will open a separate PR to fix the `markus` CLI.

**Comment by jeromesimeon** [comment]
Fix to CLI is in #118

**Comment by jeromesimeon** [comment]
I don't think I feel that strongly about this issue based on the latest fact finding. I'm closing this. If someone feels strongly that this is the right approach, feel free to reopen it.
